app/deployment/app.go

```diff
@@ -401,16 +424,30 @@ func (a *app) doDeliverCloseTx(ctx apptypes.Context, tx *types.TxCloseDeployment
 		}
 	}
 
+	groups, err := a.State().DeploymentGroup().ForDeployment(deployment.Address)
+	if err != nil {
+		return tmtypes.ResponseDeliverTx{
+			Code: code.INVALID_TRANSACTION,
+			Log:  err.Error(),
+		}
+	}
+	if deployment == nil {
+		return tmtypes.ResponseDeliverTx{
+			Code: code.INVALID_TRANSACTION,
+			Log:  "Deployment groups",
+		}
+	}
+
 	deployment.State = types.Deployment_CLOSING
 
-	for i, group := range deployment.Groups {
+	for i, group := range groups {
 		// begin for each group
 		if group.State != types.DeploymentGroup_CLOSING {
 			group.State = types.DeploymentGroup_CLOSING
-			deployment.Groups[i] = group
+			groups[i] = group
```

Comment:
we can use pointers (remove `gogoproto.nullable=false`) if it continues to be such a PITA/confusing.